### PR TITLE
Add query to load all overdue appointments for a facility

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.shortcut.com/simpledotorg/story/8325/add-query-to-load-all-overdue-appointments

### CODE REVIEW

**Comment by msasikanth** [comment]
Current dependencies on/for this PR:
* master
  * **PR #3870** <a href="https://app.graphite.dev/github/pr/simpledotorg/simple-android/3870" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/simpledotorg/simple-android/3870?utm_source=stack-comment).

File: app/src/main/java/org/simple/clinic/home/overdue/OverdueAppointmentNew.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import org.simple.clinic.overdue.Appointment
+import org.simple.clinic.overdue.callresult.CallResult
+import org.simple.clinic.patient.Gender
+import org.simple.clinic.patient.PatientAgeDetails
+import org.simple.clinic.patient.PatientPhoneNumber
+import java.time.LocalDate
+import java.util.UUID
+
+@Parcelize
+data class OverdueAppointmentNew(
```
Comment (janhavisinghh) [inline app/src/main/java/org/simple/clinic/home/overdue/OverdueAppointmentNew.kt:17]: Is there a reason why we did we not go with `OverdueAppointment_Old` and `OverdueAppointment` names for this like we usually do when we are creating a new version of something?

**Review by janhavisinghh — CHANGES_REQUESTED** [review]
See inline comments

**Review by msasikanth — COMMENTED** [review]
> Is there a reason why we did we not go with `OverdueAppointment_Old` and `OverdueAppointment` names for this like we usually do when we are creating a new version of something?

Yeah, renaming it would cause a lot of changes. So went with instead creating a new data type. Once we replace all the older usages, we can delete it and rename it. Similar process, but order changes I guess 😂

**Review by janhavisinghh — APPROVED** [review]

**Pull request merged**